tests/stdlib/tmonotimes.nim

```diff
@@ -4,17 +4,30 @@ discard """
 
 import std/[monotimes, times]
 
-let d = initDuration(nanoseconds = 10)
-let t1 = getMonoTime()
-let t2 = t1 + d
+template main =
+  block:
+    let d = initDuration(nanoseconds = 10)
+    let t1 = getMonoTime()
+    let t2 = t1 + d
 
-doAssert t2 - t1 == d
-doAssert t1 == t1
-doAssert t1 != t2
-doAssert t2 - d == t1
-doAssert t1 < t2
-doAssert t1 <= t2
-doAssert t1 <= t1
-doAssert not(t2 < t1)
-doAssert t1 < high(MonoTime)
-doAssert low(MonoTime) < t1
+    doAssert t2 - t1 == d
+    doAssert t1 == t1
+    doAssert t1 != t2
+    doAssert t2 - d == t1
+    doAssert t1 < t2
+    doAssert t1 <= t2
+    doAssert t1 <= t1
+    doAssert not(t2 < t1)
+    doAssert t1 < high(MonoTime)
+    doAssert low(MonoTime) < t1
+
+  block:
+    const n = when defined(js): 20000 else: 1000000 # keep test under ~ 1sec
+    for i in 0..<n:
+      # this could fail with getTime instead of getMonoTime, as expected
+      let a = getMonoTime()
+      let b = getMonoTime()
+      doAssert b > a
```

Comment:
> this test made no sense
So I guess the `getMonoTime` call is so cheap, then you cannot make sure b is not equal to a(at the current time precision of `getMonoTime`). 

```
Output:
tmonotimes.nim(30)       tmonotimes
assertions.nim(68)       failedAssertImpl
assertions.nim(58)       raiseAssert
fatal.nim(53)            sysFatal
Error: unhandled exception: tmonotimes.nim(30, 16) `b`gensym0 > a`gensym0`  [AssertionDefect]
```

Reply:
even if cheap, it should work by definition of what monotime is (strictly increasing), so i guess it's a bug, just filed https://github.com/nim-lang/Nim/issues/18158

Reply:
updated accordingly